exercises/simple-linked-list/Example.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+public class SimpleLinkedList<T>
+{
+    public SimpleLinkedList(T value) :
+        this(value, null)
+    {
+    }
+
+    public SimpleLinkedList(T value, SimpleLinkedList<T> next)
+    {
+        Value = value;
+        Next = next;
+    }
+
+    public T Value { get; }
+
+    public SimpleLinkedList<T> Next { get; }
+
+    public static SimpleLinkedList<T> FromEnumerable(IEnumerable<T> values)
```

Comment:
Valid point about the indices. `ICollection` might indeed be a better choice. As for the `Head`, `Push`, `Pop` and `Peek` methods, I actually think they don't belong on the list. If you look at the exercise, it says:

> As a first take, lets create a singly linked list to contain the range (1..10), and provide functions to reverse a linked list and convert to and from arrays.

The `Head`, `Push`, `Pop` and `Peek` can be part of e.g. a queue or a stack, which uses a linked list as the underlying data type, but don't necessarily belong to the linked list themselves.